### PR TITLE
fix(tests): correct metadata for algebraic and edge case tests

### LINKED ISSUE #115: Algebraic property tests still use composite function names in functions array
## Problem

Issue #60 fixed composite function mapping for `round_trip`, `load`, and `canonical_format` (via PR #61), but the same problem remains for the three algebraic property validations:

- `compose_associative` tests have `"functions": ["compose_associative"]`
- `identity_left` tests have `"functions": ["identity_left"]`
- `identity_right` tests have `"functions": ["identity_right"]`

These are not standalone functions — they are validations that require `parse`, `compose`, and `build_hierarchy`. The `functions` array should list the actual required functions, just like was done for `round_trip` → `["parse", "print"]`.

## Current state in generated_tests/

```json
{
  "name": "semigroup_associativity_basic_compose_associative",
  "validation": "compose_associative",
  "functions": ["compose_associative"],
  ...
}
```

## Expected

```json
{
  "name": "semigroup_associativity_basic_compose_associative",
  "validation": "compose_associative",
  "functions": ["parse", "compose", "build_hierarchy"],
  ...
}
```

Same change needed for `identity_left` and `identity_right` tests.

## Impact

Implementations that declare `compose` (not `compose_associative`) in their `ccl-config.yaml` filter out these tests, since the function check requires all listed functions to be supported. The TypeScript test runner works around this with a `COMPOSITE_FUNCTIONS` mapping, but new implementations hit this as a confusing failure.

## Affected files

All tests in `property_algebraic.json` with validation `compose_associative`, `identity_left`, or `identity_right`.

### LINKED ISSUE #116: Test metadata: multi-line key tests tagged with 'empty_keys' feature
## Problem

The tests `key_with_newline_before_equals` and `complex_multi_newline_whitespace` in `api_edge_cases.json` are tagged with `features: ["empty_keys"]`.

These tests exercise **multi-line key accumulation**, not empty keys. For example, `"key \n= val"` produces `{key: "key", value: "val"}` — the key is `"key"`, not empty.

The `= val` syntax *looks* like an empty key list item, but in context the `=` serves as the delimiter for the accumulated multi-line key. Tagging these as `empty_keys` is misleading.

## Suggestion

Consider adding a `multi_line_keys` feature tag (or similar) and using it for these tests instead of / in addition to `empty_keys`.

## Related

See CatConfLang/ccl-typescript#33 for the corresponding documentation gap around multi-line keys.

### PR DESCRIPTION
**Pull request by tylerbutler**
## Summary

Fixes two test data correctness issues.

* fix(generator): add algebraic validations to compositeFunctionMap

The `compose_associative`, `identity_left`, and `identity_right` validations require the `parse` and `compose` functions, but their generated `functions` arrays incorrectly listed the composite validation names instead. This caused implementations declaring `compose` support to filter out these tests. Adds the three missing entries to `compositeFunctionMap` in `generator.go`, matching the pattern established for `round_trip` in #61.

Closes #115.

* fix(tests): replace empty_keys with multiline feature tag on multi-line key tests

The `key_with_newline_before_equals` and `complex_multi_newline_whitespace` tests exercise multi-line key accumulation (key spans newlines before the `=` delimiter), not empty key handling. Both produce non-empty keys. Replaced the misleading `empty_keys` tag with `multiline`.

Closes #116.

### CODE REVIEW

File: docs/test-selection-guide.md [hunk 1]
```diff
@@ -31,6 +31,8 @@ Direct array containing CCL functions required for a test to run:
 | `get_list` | List value access | `GetList(obj, "items")` |
 | `pretty_print` | Formatting output | `PrettyPrint(obj)` |
 
+> **Note:** `build_hierarchy` internally requires `parse_indented` to produce its input entries — it cannot work with `parse` alone. `parse_indented` handles indentation-based nesting and lines without `=` delimiters that `parse` rejects. Implementations that support `build_hierarchy` implicitly support `parse_indented`, even if `parse_indented` is not exposed as a public API.
```
Comment (tylerbutler) [inline docs/test-selection-guide.md]:
```suggestion
> [!NOTE]
> `build_hierarchy` internally requires `parse_indented` to produce its input entries — it cannot work with `parse` alone. `parse_indented` handles indentation-based nesting and lines without `=` delimiters that `parse` rejects. Implementations that support `build_hierarchy` implicitly support `parse_indented`, even if `parse_indented` is not exposed as a public API.
```

File: README.md [hunk 2]
```diff
@@ -55,11 +55,14 @@ function shouldRun(test, myCapabilities) {
 | Function                                                     | Description                      |
 | ------------------------------------------------------------ | -------------------------------- |
 | `parse`                                                      | Basic key-value parsing          |
+| `parse_indented`                                             | Indentation-normalized parsing   |
 | `build_hierarchy`                                            | Object construction from entries |
 | `get_string`, `get_int`, `get_bool`, `get_float`, `get_list` | Typed value access               |
 | `filter`, `compose`                                          | Entry processing                 |
 | `canonical_format`, `round_trip`                             | Formatting and validation        |
 
+> **Note:** `build_hierarchy` internally requires `parse_indented` to produce its input entries — it cannot work with `parse` alone. `parse_indented` handles indentation-based nesting and lines without `=` delimiters that `parse` rejects. Implementations that support `build_hierarchy` implicitly support `parse_indented`, even if `parse_indented` is not exposed as a public API.
```
Comment (tylerbutler) [inline README.md]:
```suggestion
> [!NOTE]
> `build_hierarchy` internally requires `parse_indented` to produce its input entries — it cannot work with `parse` alone. `parse_indented` handles indentation-based nesting and lines without `=` delimiters that `parse` rejects. Implementations that support `build_hierarchy` implicitly support `parse_indented`, even if `parse_indented` is not exposed as a public API.
```

**Comment by tylerbutler** [comment]
Split into #124 (fix), #125 (docs), and #126 (feat). The original feat portion of this PR was obsoleted by #121 landing on main; only the new multiline key test cases remain, which #126 contributes.
